### ISSUE TITLE
How can i filter on multiple values?

### ISSUE THREAD

**Issue by rahul1059** [AllenFang/react-bootstrap-table#1202]
I am using handleFilterData for custom filters. I have checkboxes in select dropdown, and can select multiple values from dropdown. If i pass mutiple values as array e.g. ["All","onlyOne"] to handleFilterData, its not working. However if i pass only one value it works fine like ["All".

Do you have any solution for this ?

**Comment by sunnybogawat** [comment]
I am also facing the same issue and want to know is there any way to pass an array of values instead of single value.

**Comment by AllenFang** [comment]
ok, I'll try to support it, but It may need some time

**Comment by jsangilve** [comment]
Hi @AllenFang, have you made any progress on this issue? 
I've been reading the code and I believe all changes required to support this feature should be made on `TableDataStore`'s methods? I could help you to do this.

**Comment by AllenFang** [comment]
sorry, I'm busy recently, PR is welcome, feel free to discuss with me if you have any questions.

**Comment by AllenFang** [comment]
BTW, I'll review this issues and try to implement it again in near future.

**Comment by bpsourav21** [comment]
Hello guyz,
Somehow I managed to solve similar problem as you asked.I have used custom filter. And modified this filter as a select tag. In "Select Filter" it has an issue. It always required a object which is designed like below:
option:{0: "Rice", 1:"Pants", 2:"Shirts"}

But I have modified them using custom filter where I can use a list of array.

the demo code is given below:
bootstrap table code:
```

getCustomFilter(filterHandler, options) {
        return (
            <CustomFilter filterHandler={filterHandler} />
        );
    }
  render() {
 return (
<div>
<BootstrapTable data={this.props.all_products} search={true} pagination={true}>
                <TableHeaderColumn width="25%" dataField='product_code' isKey={true}>Product Code</TableHeaderColumn>
                <TableHeaderColumn width="25%" dataField='product_name'>Product Name</TableHeaderColumn>
                <TableHeaderColumn width="15%" dataField={this.state.selectField}>Product Name</TableHeaderColumn>
                <TableHeaderColumn width="15%" dataField="category" filter={{ type: 'CustomFilter', getElement: this.getCustomFilter.bind(this) }}>Category</TableHeaderColumn>
                <TableHeaderColumn width="20%" dataField='createdAt' dataFormat={this.dateFormatter.bind(this)} >Date</TableHeaderColumn>
            </BootstrapTable>
</div>
);



```
CustomFilter.js



    dataSelected(filterHandler, e) {
        var value = e.target.value
        if (value == "Select") {
            filterHandler("")
        }
        else {
            filterHandler(value)
        }
    }

    render() {
        return (
            <select name="selectData" className="form-control" onChange={this.dataSelected.bind(this, this.props.filterHandler)}>
                <option value="Select" >Select</option>
                <option value="Rice">Rice</option>
                <option value="Shirt">Shirt</option>
                <option value="pants">pants</option>
            </select>
        );
    }



=================================================================

here a screenshot is given of following work:
![untitled](https://user-images.githubusercontent.com/19387437/28753011-58cdb72a-754e-11e7-8d63-2eadde3b1d6c.png)

**Comment by AllenFang** [comment]
Fixed on https://github.com/AllenFang/react-bootstrap-table/pull/1624  and released on `v4.0.5`
